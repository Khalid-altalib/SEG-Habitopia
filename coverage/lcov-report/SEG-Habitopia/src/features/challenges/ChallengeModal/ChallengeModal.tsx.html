
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for SEG-Habitopia/src/features/challenges/ChallengeModal/ChallengeModal.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../../index.html">All files</a> / <a href="index.html">SEG-Habitopia/src/features/challenges/ChallengeModal</a> ChallengeModal.tsx</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>15/15</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>1/2</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>4/4</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>15/15</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">29x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// Import necessary modules
import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { HStack, View, VStack, Image } from "native-base";
import React from "react";
import { ButtonType, Challenge, RootParams, TextType } from "../../../../types";
import { useDispatch, useSelector } from "../../../app/hooks";
import { challengeMappings } from "../challengeMappings";
import { joinChallenge } from "../challengesSlice";
import { fetchChats } from "@features/chat/chatSlice";
import Background from "@components/Background";
import Text from "@components/Text";
import Button from "@components/Button";
&nbsp;
type Props = {
  challenge: Challenge;
};
&nbsp;
const ChallengeModal = (props: Props) =&gt; {
  const { challenge } = props;
  const { name, description } = challenge;
&nbsp;
  // Get the joinChallenge request status from the Redux store
  const { joinChallenge: requestStatus } = useSelector(
    (state) =&gt; state.challenges
  );
&nbsp;
  // Extract the loading status from the requestStatus object
  const { loading } = requestStatus;
&nbsp;
  // Get the image URI for the challenge from the challengeMappings object
  const { image } = challengeMappings[name] || <span class="branch-1 cbranch-no" title="branch not covered" >challengeMappings["fallback"];</span>
&nbsp;
  // Get the navigation object
  const navigation = useNavigation&lt;NativeStackNavigationProp&lt;RootParams&gt;&gt;();
&nbsp;
  const dispatch = useDispatch();
&nbsp;
  // Define a function to handle the button click event
  const handleButtonClick = async () =&gt; {
    // Dispatch the joinChallenge action with the name of the challenge
    await dispatch(joinChallenge(name));
    // Dispatch the fetchChats action to update the chats in the store
    dispatch(fetchChats());
    // Pop the current screen off the navigation stack to go back to the previous screen
    navigation.pop();
  };
&nbsp;
  // Render the component
  return (
    &lt;View testID="challengePrompt"&gt;
      &lt;Background&gt;
        &lt;VStack
          space={3}
          alignItems="center"
          justifyContent="center"
          height="100%"
          padding={20}
        &gt;
          {/* Render the name of the challenge as a heading */}
          &lt;Text type={TextType.Heading}&gt;{name}&lt;/Text&gt;
&nbsp;
          {/* Render the image for the challenge */}
          &lt;Image
            width={150}
            height={150}
            alt="Challenge Image"
            borderRadius="lg"
            source={{ uri: image }}
          /&gt;
&nbsp;
          &lt;Text type={TextType.Regular}&gt;{description}&lt;/Text&gt;
&nbsp;
          &lt;VStack space={5} marginTop={20}&gt;
            &lt;Button
              type={ButtonType.Primary}
              onPress={handleButtonClick}
              isLoading={loading}
              isFullWidth
            &gt;
              Join Challenge!
            &lt;/Button&gt;
            &lt;Button
              type={ButtonType.Secondary}
              onPress={() =&gt; navigation.goBack()}
              isFullWidth
            &gt;
              Not now
            &lt;/Button&gt;
          &lt;/VStack&gt;
        &lt;/VStack&gt;
      &lt;/Background&gt;
    &lt;/View&gt;
  );
};
&nbsp;
// Export the component as the default export
export default ChallengeModal;
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2023-03-29T22:54:44.925Z
            </div>
        <script src="../../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../../sorter.js"></script>
        <script src="../../../../../block-navigation.js"></script>
    </body>
</html>
    