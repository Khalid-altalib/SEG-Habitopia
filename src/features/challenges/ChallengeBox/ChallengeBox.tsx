import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { Box, Heading, HStack } from "native-base";
import React from "react";
import { TouchableOpacity, Image } from "react-native";
import { Challenge, RootParams } from "../../../../types";
import { challengeMappings } from "../challengeMappings";
import ChallengeOnGoingText from "../ChallengeOnGoingText/ChallengeOnGoingText";

type Props = {
  challenge: Challenge;
};

const ChallengeBox = (props: Props) => {
  const { challenge } = props;
  const { name, active } = challenge;

  const navigation = useNavigation<NativeStackNavigationProp<RootParams>>();

  const { color: cardColor, image: cardImage } =
    challengeMappings[name] || challengeMappings["fallback"];

  const handlePress = () =>
    navigation.navigate("ChallengePrompt", {
      challenge: challenge,
    });

  return (
    <TouchableOpacity onPress={handlePress}>
      <Box
        backgroundColor={cardColor}
        borderRadius={"md"}
        shadow={3}
        padding={3}
        height={100}
      >
        <HStack justifyContent="space-between" height="100%">
          <Box justifyContent="space-between">
            <Heading>{name}</Heading>
            <ChallengeOnGoingText onGoing={active} />
          </Box>
          <Box borderRadius={3}>
            <Image
              source={{ uri: cardImage }}
              borderRadius={6}
              style={{ height: "100%", width: 80 }}
            />
          </Box>
        </HStack>
      </Box>
    </TouchableOpacity>
  );
};

export default ChallengeBox;
