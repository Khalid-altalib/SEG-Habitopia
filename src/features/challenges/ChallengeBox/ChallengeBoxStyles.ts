import { background, border } from "native-base/lib/typescript/theme/styled-system";
import { StyleSheet } from "react-native";

const challengeBoxStyles = StyleSheet.create({
unactiveChallengeBox: {
    flex: 1,
    width: '100%',
    overflow: 'hidden',
    margin: 5,
    borderRadius: 20,
    shadowColor: '#000',
    justifyContent: 'center',
    shadowOffset: {
        width: 0,
        height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
    borderWidth: 4,
    minHeight:80,
},
activeChallengeBox:{
    flex: 1,
    width: '100%',
    overflow: 'hidden',
    margin: 5,
    borderRadius: 20,
    shadowColor: '#000',
    justifyContent: 'center',
    shadowOffset: {
        width: 0,
        height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
    borderColor: 'red',
    borderWidth: 4,
    minHeight:80,
},
challengeText: {
    fontSize: 25,
},
onGoingText: {
    color: 'red'
},
challengeImage: {
    height: 80,
    width: 80,
    marginRight: 0,
    marginLeft: 'auto',
}
});

export default challengeBoxStyles;