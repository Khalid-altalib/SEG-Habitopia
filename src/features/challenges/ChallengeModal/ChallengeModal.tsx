import { RouteProp, useNavigation, useRoute } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { isLoading } from "expo-font";
import {
  Button,
  HStack,
  Text,
  View,
  Alert,
  Modal,
  VStack,
  Image,
  Heading,
} from "native-base";
import React from "react";
import { Challenge, RootParams } from "../../../../types";
import { useDispatch, useSelector } from "../../../app/hooks";
import { challengeMappings } from "../challengeMappings";
import { joinChallenge } from "../challengesSlice";

type Props = {
  challenge: Challenge;
};

const ChallengeModal = (props: Props) => {
  const { challenge } = props;
  const { name, description } = challenge;
  const { joinChallenge: requestStatus } = useSelector(
    (state) => state.challenges
  );
  const { loading } = requestStatus;

  const { image } = challengeMappings[name] || challengeMappings["fallback"];

  const navigation = useNavigation<NativeStackNavigationProp<RootParams>>();

  const dispatch = useDispatch();

  const handleButtonClick = async () => {
    await dispatch(joinChallenge(name));
    navigation.pop();
  };

  return (
    <View>
      <VStack space={3} alignItems="center">
        <Heading size="3xl">{name}</Heading>
        <Image
          width={100}
          height={100}
          alt="Challenge Image"
          borderRadius="lg"
          source={{ uri: image }}
        />
        <Text fontSize="xl" textAlign="center">
          {description}
        </Text>

        <HStack space={3}>
          <Button
            backgroundColor="amber.500"
            onPress={handleButtonClick}
            isLoading={loading}
          >
            Join Challenge!
          </Button>
          <Button
            backgroundColor="gray.400"
            onPress={() => navigation.goBack()}
          >
            Not now
          </Button>
        </HStack>
      </VStack>
    </View>
  );
};

export default ChallengeModal;
