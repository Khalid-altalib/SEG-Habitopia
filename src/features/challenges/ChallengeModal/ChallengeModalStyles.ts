import { StyleSheet } from "react-native";

const challengeModalStyles = StyleSheet.create({
  challengeModal: {
    margin: 20,
    backgroundColor: "white",
    borderRadius: 20,
    padding: 35,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
});

export default challengeModalStyles;
