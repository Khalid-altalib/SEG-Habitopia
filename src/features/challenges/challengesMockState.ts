import { ChallengesState } from "./challengesSlice";

// Mocks the challenge state for testing purposes
const challengesMockState: ChallengesState = {
    challenges: [
        {
            id: '1',
            name: 'Fitness',
            description: 'Be active',
            active: true

        },
        {
            id: '1',
            name: 'Study',
            description: 'Study hard',
            active: false
        }
    ],
    fetchChallenges: {
        loading: false,
        error: ""
    },
    joinChallenge: {
        loading: false,
        error: ""
    }
};

export default challengesMockState;