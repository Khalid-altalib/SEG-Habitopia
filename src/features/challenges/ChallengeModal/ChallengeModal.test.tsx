import { fireEvent, render, screen } from "@testing-library/react-native";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import TestingWrapperNavigation from "@app/testingWrapperWithNavigation";
import challengesMockState from "../challengesMockState";
import { extendTheme, NativeBaseProvider } from "native-base";
import ChallengeModal from "./ChallengeModal";
import ChallengeBox from "../ChallengeBox/ChallengeBox";

describe("ChallengeModal", () => {
  const mockState = {
    challenges: challengesMockState,
    joinChallenge: jest.fn(),
  };
  const mockStore = configureStore([thunk])(mockState);

  const modalProps = {
    challenge: mockState.challenges.challenges[0],
  };

  it("renders the challenge modal correctly", async () => {
    const component = render(
      <TestingWrapperNavigation store={mockStore}>
        <ChallengeModal {...modalProps} />
      </TestingWrapperNavigation>
    );

    expect(component).toMatchSnapshot();
  });

  it("finds the challenge modal", async () => {
    const component = render(
      <TestingWrapperNavigation store={mockStore}>
        <ChallengeModal {...modalProps} />
      </TestingWrapperNavigation>
    );

    const box = component.getByTestId("challengePrompt");

    expect(box).toBeDefined();
  });

  it("dispatches a join request to join button is pressed", async () => {
    const component = render(
      <TestingWrapperNavigation store={mockStore}>
        <ChallengeModal {...modalProps} />
      </TestingWrapperNavigation>
    );

    const button = component.getByTestId("joinButton");

    fireEvent.press(button);

    expect(mockStore.getActions()[0].type).toEqual("challenges/join/pending");
  });

  it("navigates back when back button is pressed", async () => {
    const theme = extendTheme({});
    const inset = {
      frame: { x: 0, y: 0, width: 0, height: 0 },
      insets: { top: 0, left: 0, right: 0, bottom: 0 },
    };
    const component = render(
      <TestingWrapperNavigation store={mockStore}>
        <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
          <ChallengeBox {...modalProps} />
        </NativeBaseProvider>
      </TestingWrapperNavigation>
    );

    const box = component.getByTestId("challengeBox");
    fireEvent(box, "press");

    const backButton = component.getByTestId("backButton");
    fireEvent.press(backButton);

    const checkChallengePrompt = screen.queryByTestId("challengePrompt");
    expect(checkChallengePrompt).toBeNull();
  });
});
