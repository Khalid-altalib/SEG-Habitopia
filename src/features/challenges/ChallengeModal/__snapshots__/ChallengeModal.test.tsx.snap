// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChallengeModal renders the challenge modal correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="TestScreen"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              dataSet={{}}
              style={{}}
              testID="challengePrompt"
            >
              <View
                dataSet={{}}
                style={
                  {
                    "alignItems": "center",
                  }
                }
              >
                <Text
                  dataSet={{}}
                  style={
                    {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 48,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "letterSpacing": 2.4000000000000004,
                      "lineHeight": 60,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Fitness
                </Text>
                <View
                  dataSet={{}}
                  style={
                    {
                      "height": 12,
                    }
                  }
                />
                <Image
                  accessibilityLabel="Challenge Image"
                  alt="Challenge Image"
                  dataSet={{}}
                  onError={[Function]}
                  source={
                    {
                      "uri": "https://picsum.photos/200",
                    }
                  }
                  style={
                    {
                      "borderRadius": 8,
                      "height": 100,
                      "maxWidth": "100%",
                      "width": 100,
                    }
                  }
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "height": 12,
                    }
                  }
                />
                <Text
                  dataSet={{}}
                  style={
                    {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 20,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 30,
                      "textAlign": "center",
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Be active
                </Text>
                <View
                  dataSet={{}}
                  style={
                    {
                      "height": 12,
                    }
                  }
                />
                <View
                  dataSet={{}}
                  style={
                    {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    collapsable={false}
                    colorScheme="primary"
                    dataSet={{}}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#f59e0b",
                        "borderRadius": 4,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "paddingBottom": 10,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 10,
                      }
                    }
                    testID="joinButton"
                  >
                    <View
                      dataSet={{}}
                      style={
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                        }
                      }
                      test={true}
                    >
                      <View
                        dataSet={{}}
                        style={{}}
                      >
                        <Text
                          dataSet={{}}
                          style={
                            {
                              "backgroundColor": undefined,
                              "color": "#fafafa",
                              "fontFamily": undefined,
                              "fontSize": 14,
                              "fontStyle": "normal",
                              "fontWeight": "400",
                              "letterSpacing": 0,
                              "lineHeight": 21,
                              "textDecorationLine": undefined,
                            }
                          }
                        >
                          Join Challenge!
                        </Text>
                      </View>
                    </View>
                  </View>
                  <View
                    dataSet={{}}
                    style={
                      {
                        "width": 12,
                      }
                    }
                  />
                  <View
                    accessibilityRole="button"
                    accessible={true}
                    collapsable={false}
                    colorScheme="primary"
                    dataSet={{}}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#a1a1aa",
                        "borderRadius": 4,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "paddingBottom": 10,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 10,
                      }
                    }
                    testID="backButton"
                  >
                    <View
                      dataSet={{}}
                      style={
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                        }
                      }
                      test={true}
                    >
                      <View
                        dataSet={{}}
                        style={{}}
                      >
                        <Text
                          dataSet={{}}
                          style={
                            {
                              "backgroundColor": undefined,
                              "color": "#fafafa",
                              "fontFamily": undefined,
                              "fontSize": 14,
                              "fontStyle": "normal",
                              "fontWeight": "400",
                              "letterSpacing": 0,
                              "lineHeight": 21,
                              "textDecorationLine": undefined,
                            }
                          }
                        >
                          Not now
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
